pkg/graph/resolvers/mto_solution_test.go

```diff
@@ -157,3 +158,50 @@ func (suite *ResolverSuite) TestCreateCommonSolutionAndLinkMilestones() {
 	suite.Len(milestonesAfterDelete, 0, "After deleting the solution, there should be no linked milestones")
 
 }
+
+func (suite *ResolverSuite) TestMTOSolutionUpdateLinkedMilestoness_AddByMilestoneID() {
+	plan := suite.createModelPlan("plan for adding milestones by milestone ID")
+
+	// Create a milestone with no linked solutions
+	milestone, err := MTOMilestoneCreateCustom(suite.testConfigs.Context, suite.testConfigs.Logger, suite.testConfigs.Principal, suite.testConfigs.Store, "Initial Milestone", plan.ID, nil)
+	suite.NoError(err)
+
+	// Create a custom solution
+	solName := "Custom Solution 1"
+	solType := models.MTOSolutionTypeOther
+	sol, err := MTOSolutionCreateCustom(
+		suite.testConfigs.Logger,
+		suite.testConfigs.Principal,
+		suite.testConfigs.Store,
+		plan.ID,
+		nil,
+		solName,
+		solType,
+		nil,
+		"POC Name",
+		"poc@example.com",
+	)
+	suite.NoError(err)
+
+	// Add the solution to the milestone by solution ID
+	milestoneLinks := &model.MTOMilestoneLinks{
+		MilestoneIDs: []uuid.UUID{milestone.ID},
+	}
+
+	_, err = MTOSolutionUpdate(
+		suite.testConfigs.Context,
+		suite.testConfigs.Logger,
+		suite.testConfigs.Principal,
+		suite.testConfigs.Store,
+		sol.ID,
+		map[string]interface{}{},
+		milestoneLinks,
+	)
+	suite.NoError(err)
+
+	// Verify the solution is linked now
+	linkedSolutions, err := storage.MTOMilestoneSolutionLinkGetByMilestoneID(suite.testConfigs.Store, suite.testConfigs.Logger, milestone.ID)
+	suite.NoError(err)
+	suite.Len(linkedSolutions, 1)
+	suite.Equal(sol.ID, linkedSolutions[0].SolutionID)
+}
```

Comment:
Since this resolver also is capable of unlinking solutions, this should probably test the unlinking behavior, as well as the behavior when the links input is not provided.

Reply:
Fair enough